src/components/vive-controls.js

```diff
@@ -43,36 +45,129 @@ module.exports.Component = registerComponent('vive-controls', {
     this.onButtonDown = function (evt) { self.onButtonEvent(evt.detail.id, 'down'); };
     this.onButtonUp = function (evt) { self.onButtonEvent(evt.detail.id, 'up'); };
     this.onModelLoaded = bind(this.onModelLoaded, this);
+    this.controllerPresent = false;
+    this.everGotGamepadEvent = false;
+    this.lastControllerCheck = 0;
   },
 
-  play: function () {
+  startListening: function () {
     var el = this.el;
+    // console.log('startListening');
     el.addEventListener('buttonchanged', this.onButtonChanged);
     el.addEventListener('buttondown', this.onButtonDown);
     el.addEventListener('buttonup', this.onButtonUp);
     el.addEventListener('model-loaded', this.onModelLoaded);
   },
 
-  pause: function () {
+  stopListening: function () {
     var el = this.el;
+    // console.log('stopListening');
     el.removeEventListener('buttonchanged', this.onButtonChanged);
     el.removeEventListener('buttondown', this.onButtonDown);
     el.removeEventListener('buttonup', this.onButtonUp);
     el.removeEventListener('model-loaded', this.onModelLoaded);
   },
 
-  update: function () {
+  checkIfControllerPresent: function () {
```

Comment:
(same as above)

OK, made `enumerateGamepads` in utils, and changed `tracked-controls` to use it too... but didn't move `checkIfControllerPresent` into utils since it does differ between `oculus-touch-controls` and `vive-controls`